lib/luminork-server/src/service/v1/schemas/install_schema.rs

```diff
@@ -0,0 +1,80 @@
+use axum::{
+    extract::Path,
+    response::Json,
+};
+use dal::{
+    Schema,
+    SchemaVariant,
+};
+use itertools::Itertools;
+use sdf_extract::{
+    PosthogEventTracker,
+    change_set::ChangeSetDalContext,
+};
+use serde_json::json;
+use utoipa::{
+    self,
+};
+
+use super::{
+    GetSchemaV1Response,
+    SchemaError,
+    SchemaResult,
+    SchemaV1RequestPath,
+};
+
+#[utoipa::path(
+    post,
+    path = "/v1/w/{workspace_id}/change-sets/{change_set_id}/schemas/{schema_id}/install",
+    params(
+        ("workspace_id" = String, Path, description = "Workspace identifier"),
+        ("change_set_id" = String, Path, description = "Change Set identifier"),
+        ("schema_id" = String, Path, description = "Schema identifier"),
+    ),
+    tag = "schemas",
+    summary = "Installs a schema - if there's an installed schema, it will return that schema detail",
+    responses(
+        (status = 200, description = "Schema installed successfully", body = GetSchemaV1Response),
+        (status = 401, description = "Unauthorized - Invalid or missing token"),
+        (status = 422, description = "Validation error - Invalid request data", body = crate::service::v1::common::ApiError),
+        (status = 500, description = "Internal server error", body = crate::service::v1::common::ApiError)
+    )
+)]
+pub async fn install_schema(
+    ChangeSetDalContext(ref ctx): ChangeSetDalContext,
+    tracker: PosthogEventTracker,
+    Path(SchemaV1RequestPath { schema_id }): Path<SchemaV1RequestPath>,
+) -> SchemaResult<Json<GetSchemaV1Response>> {
+    if ctx.change_set_id() == ctx.get_workspace_default_change_set_id().await? {
+        return Err(SchemaError::NotPermittedOnHead);
+    }
+
+    let schema = if let Some(installed_schema) = Schema::get_by_id_opt(ctx, schema_id).await? {
+        installed_schema
+    } else {
+        Schema::get_or_install_default_variant(ctx, schema_id).await?;
+        Schema::get_by_id_opt(ctx, schema_id)
+            .await?
+            .ok_or(SchemaError::SchemaNotFound(schema_id))?
```

Comment:
Not blocking, but why not use `get_by_id`? Is the conversion here better for error handling?